tests/integration/long/test_ssl.py

```diff
@@ -108,4 +132,109 @@ def test_can_connect_with_ssl_ca(self):
         statement.consistency_level = ConsistencyLevel.ANY
         session.execute(statement)
 
-        cluster.shutdown()
\ No newline at end of file
+        cluster.shutdown()
+
+
+class SSLConnectionAuthTests(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        setup_cluster_ssl(client_auth=True)
+
+    def test_can_connect_with_ssl_client_auth(self):
+        """
+        Test to validate that we can connect to a C* cluster that has client_auth enabled.
+
+        This test will setup and use a c* cluster that has client authentication enabled. It will then attempt
+        to connect using valid client keys, and certs (that are in the server's truststore), and attempt to preform some
+        basic operations
+        @since 2.7.0
+
+        @expected_result The client can connect via SSL and preform some basic operations
+
+        @test_category connection:ssl
+        """
+
+        # Need to get absolute paths for certs/key
+        abs_path_ca_cert_path = os.path.abspath(CLIENT_CA_CERTS)
+        abs_driver_keyfile = os.path.abspath(DRIVER_KEYFILE)
+        abs_driver_certfile = os.path.abspath(DRIVER_CERTFILE)
+
+        tries = 0
+        while True:
+            if tries > 5:
+                raise RuntimeError("Failed to connect to SSL cluster after 5 attempts")
+            try:
+                cluster = Cluster(protocol_version=PROTOCOL_VERSION, ssl_options={'ca_certs': abs_path_ca_cert_path,
+                                                                                  'ssl_version': ssl.PROTOCOL_TLSv1,
+                                                                                  'keyfile': abs_driver_keyfile,
+                                                                                  'certfile': abs_driver_certfile})
+
+                session = cluster.connect()
+                break
+            except Exception:
+                ex_type, ex, tb = sys.exc_info()
+                log.warn("{0}: {1} Backtrace: {2}".format(ex_type.__name__, ex, traceback.extract_tb(tb)))
+                del tb
+                tries += 1
+
+        # attempt a few simple commands.
```

Comment:
We should come up with a standard policy for these types of tests. I was attempting to setup the ssl tests in such a way where they will have no affect on any other tests. we have standard keyspaces for various replication factors test1rf, test2rf, test3rf.. etc. but I don't see any standard tables, perhaps we should add some for simple validations such as the above.